File: C++/greatest_common_divisor.cpp
#include <iostream>

int gcd(int a, int b) {

    if (b == 0) {
        return a;
    } else if (a == 0) {
        return b;
    } else {

        if (a > b) {
            return gcd(a - b, b);
        } else {
            return gcd(a, b - a);
        }

    }
}

int main(int argc, char *argv[]) {

    int a, b;

    std::cout << "Enter a: ";
    std::cin >> a;

    std::cout << "Enter b: ";
    std::cin >> b;

    int greatestCommonDivisor = gcd(a, b);

    std::cout << "Greatest Common Divisor: " << greatestCommonDivisor << std::endl;

    return 0;
}

File: python/greatest_common_divisor.py

def gcd(a, b):

    if b == 0:
        return a
    elif a == 0:
        return b
    else:

        if a > b:
            return gcd(a - b, b)
        else: 
            return gcd(a, b - a)


if __name__ == "__main__":
    
    a = int(input("Enter a: "))
    b = int(input("Enter b: "))

    result = gcd(a,b)

    print("Greatest Common Divisor: ", result)
